src/runtime_src/core/edge/user/aie/common_layer/adf_runtime_api.cpp

```diff
@@ -0,0 +1,710 @@
+/**
+* Copyright (C) 2020 Xilinx, Inc
+*
+* Licensed under the Apache License, Version 2.0 (the "License"). You may
+* not use this file except in compliance with the License. A copy of the
+* License is located at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+* License for the specific language governing permissions and limitations
+* under the License.
+*/
+
+#include "adf_runtime_api.h"
+#include "adf_api_message.h"
+
+#include <algorithm>
+#include <sstream>
+#include <map>
+
+extern "C"
+{
+    #include "xaiengine.h"
+}
+
+namespace adf {
+/********************************* Statics & Constants *********************************/
+
+static constexpr short INVALID_TILE_COORD = 0xFF;
+
+static constexpr int ACQ_WRITE = 0;
+static constexpr int ACQ_READ = 1;
+static constexpr int REL_READ = 1;
+static constexpr int REL_WRITE = 0;
+
+static constexpr int AIE_ML_REL_WRITE = -1;
+static constexpr int AIE_ML_ASYNC_REL = 1;
+static constexpr int AIE_ML_ASYNC_ACQ = -1; //negative lock value -> acquire_greater_equal
+static constexpr int AIE_ML_ASYNC_ACQ_FIRST_TIME = 0;
+
+static constexpr unsigned LOCK_TIMEOUT = 0x7FFFFFFF;
+
+
+/********************************* config_manager *************************************/
+
+XAie_DevInst* config_manager::s_pDevInst = nullptr;
+bool config_manager::s_bInitialized = false;
+size_t config_manager::s_num_reserved_rows = 0;
+bool config_manager::s_broadcast_enable_core = false;
+
+err_code config_manager::initialize(XAie_DevInst* devInst, size_t num_reserved_rows, bool broadcast_enable_core)
+{
+    if(!s_bInitialized)
+    {
+        if(!devInst)
+            return errorMsg(err_code::internal_error, "ERROR: config_manager::initialize: Cannot initialize device instance.");
+        
+        s_pDevInst = devInst;
+        s_num_reserved_rows = num_reserved_rows;
+        s_broadcast_enable_core = broadcast_enable_core;
+
+        /*
+        Resources::AIE::initialize(s_pDevInst->NumCols, s_pDevInst->NumRows - 1);
+        
+        debugMsg(static_cast<std::stringstream &&>(std::stringstream() << "config_manager::initialize: initialized Resources::AIE: " << (int)s_pDevInst->NumCols << " columns, " << (int)(s_pDevInst->NumRows - 1) << " AIE rows").str());
+        
+        if (s_pDevInst->EccStatus == XAIE_ENABLE)
+        {
+            if (Resources::AIE::reservePerformanceCounterEccScrubbing())
+                debugMsg("config_manager::initialize: reserved performance counter in core module for ECC Scrubbing");
+            else
+                return errorMsg(err_code::internal_error, "ERROR: config_manager::initialize: failed to reserve performance counter in core module for ECC Scrubbing");
+        }*/
```

Comment:
Comment should be
//
//
or 
/*
 *
 */
 
if it is dead code, we can remove it